### PR TITLE
add User controller, model and routes

### PR DESCRIPTION
**Pull request by Tanyaa-a**
## Description
 Implemented basic setup for back end.  It includes a User model, controller, and routes for user registration. It also organizes the project with main folders for controllers, models, routes, errors and database configuration.

## Changes
- Added User model with fields for firstName, lastName, email, password, and role.
- Created User controller for handling user registration.
- Set up routes for user registration (/register endpoint).
- Organized project structure with main folders.

### CODE REVIEW

File: src/models/User.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+      'Please provide a valid email',
+    ],
+  },
+  password: {
+    type: String,
+    required: [true, 'Please provide password'],
+    minlength: 6,
+  },
+  role: {
```
Comment (vrudnitskaya) [inline src/models/User.js]: If I'm not mistaken we decided not to specify any role. Every user can do both sell and purchase.

File: src/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    await newUser.save();
+    console.log('User registered:', newUser);
+
+    res.status(201).json({
+      message: 'User registered successfully',
+      user: { id: newUser._id, email: newUser.email },
+    });
+  } catch (error) {
+    console.error('Error registering user:', error);
+    res.status(500).json({ message: 'Internal server error' });
```
Comment (vrudnitskaya) [inline src/controllers/userController.js]: Please install statusCode package to handle the statuses

File: src/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const registerUser = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+
+    const existingUser = await User.findOne({ email });
+    if (existingUser) {
+      return res.status(400).json({ message: 'User already exists' });
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (vrudnitskaya) [inline src/controllers/userController.js]: I think it's better to move the hashing password functionality to the Model

**Review by vrudnitskaya — CHANGES_REQUESTED** [review]
Good job, Tanya! I was able to register the user. 

Also please add the login and logout functionality  to finish the implementing the user's routes and controllers.

register

File: src/models/User.js (hunk 1 above)
Comment (Tanyaa-a) [reply src/models/User.js]: I will add location then and the seller's books we'll add when the booksSchema will be ready

File: src/routes/userRouter.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const router = express.Router();
+const userController = require('../controllers/userController.js');
```
Comment (vrudnitskaya) [inline src/routes/userRouter.js:3]: You can destructure the user controller while importing it.

**Review by vrudnitskaya — APPROVED** [review]
Nice work, Tanya. I've successfully registered a new user. Also the login and logout functionality work as they should.

**Review by benochi — APPROVED** [review]

**Pull request merged**